polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1381,6 +1434,96 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       }
     }
 
+    /**
+     * COPIED FROM {@link BaseMetastoreTableOperations} but without the requirement that base ==
+     * current()
+     *
+     * @param base table metadata on which changes were based
+     * @param metadata new table metadata with updates
+     */
+    @Override
+    public void commit(TableMetadata base, TableMetadata metadata) {
+      TableMetadata currentMetadata = current();
+
+      // if the metadata is already out of date, reject it
+      if (base == null) {
+        if (currentMetadata != null) {
+          // when current is non-null, the table exists. but when base is null, the commit is trying
+          // to create the table
+          throw new AlreadyExistsException("Table already exists: %s", tableName());
+        }
+      } else if (base.metadataFileLocation() != null
+          && !base.metadataFileLocation().equals(currentMetadata.metadataFileLocation())) {
+        throw new CommitFailedException("Cannot commit: stale table metadata");
+      } else if (base != currentMetadata) {
+        // This branch is different from BaseMetastoreTableOperations
+        LOGGER.debug(
+            "Base object differs from current metadata; proceeding because locations match");
+      } else if (base.metadataFileLocation().equals(metadata.metadataFileLocation())) {
+        // if the metadata is not changed, return early
+        LOGGER.info("Nothing to commit.");
+        return;
+      }
+
+      long start = System.currentTimeMillis();
+      doCommit(base, metadata);
+      deleteRemovedMetadataFiles(base, metadata);
+      requestRefresh();
+
+      LOGGER.info(
+          "Successfully committed to table {} in {} ms",
+          tableName(),
+          System.currentTimeMillis() - start);
+    }
+
+    /**
+     * COPIED FROM {@link BaseMetastoreTableOperations}
```

Comment:
You mean copied from `CatalogUtil`? Add comments here explaining why this is copied from there, what changes, etc

Reply:
It was [recently refactored](https://github.com/apache/iceberg/commit/c16cefa5015dda417f80e0d59124cd92448787ab); in the version of Iceberg we depend on this method is still in `BaseMetastoreTableOperations` and that is where I coped it from.

There are no changes; the method simply is private in `BaseMetastoreTableOperations` and so it needs to be copied to be called from a private class. If/when the dependency upgrade goes through, we can remove this copy. I will add a comment to that effect 👍